src/snapshot/metadata.rs

```diff
@@ -134,26 +135,27 @@ impl InstanceContext {
         rules.extend(new_rules);
     }
 
-    pub fn set_script_type(&mut self, script_type: ScriptContextType) {
-        self.script_type = script_type
+    pub fn set_emit_legacy_scripts(&mut self, emit_legacy_scripts: bool) {
+        self.emit_legacy_scripts = emit_legacy_scripts
     }
 }
 
 // serve_session always passes an option from the config file, but tests want it to be explict
 #[cfg(test)]
-impl From<ScriptContextType> for InstanceContext {
-    fn from(script_type: ScriptContextType) -> Self {
+impl From<bool> for InstanceContext {
```

Comment:
I'm going to have to veto having a `From<bool>` implementation for `InstanceContext`, I'm afraid. I would rather we have a constructor named `InstanceContext::with_legacy_emissions` or something to the effect instead.

This is because `From<T> for U` also implies `Into<U> for T` and I don't really want us to have the power to use a boolean where we need an `InstanceContext` by just using `into()`. It's asking for trouble.